stages/org.osbuild.ostree.deploy

```diff
@@ -100,51 +122,96 @@ def make_fs_identifier(desc):
     for key in ["uuid", "label"]:
         val = desc.get(key)
         if val:
-            return f"{key.upper()}={val}"
+          return f"{key.upper()}={val}"
     raise ValueError("unknown rootfs type")
 
 
 def main(tree, inputs, options):
+    # distingush between ostree commit or container image input
+    input_type = None
+    for key in inputs:
+      input_type = key
+    
+    # check if "ref" key exists to avoid accessing a non-existant key, value pair
+    if "ref" not in options.keys():
+      if not input_type or input_type == "commits":
+        raise ValueError("ref missing")
+    else:
+      ref = options["ref"]
```

Comment:
I think this is just:

```
ref = options.get("ref")
```

i.e. if it's not in the options it won't error.

Reply:
oh, I see. you want to add some verification that if it is needed we'll raise a valueError if it wasn't provided.

Reply:
Yeah, the main point of this part is to add verification in the code since we made `ref` optional in the schema, but using `ref = options.get("ref")`and checking for `None` is still better than using `"ref" not in options.keys()`